providers/implementations/signature/eddsa_sig.c

```diff
@@ -167,29 +258,75 @@ int ed25519_digest_sign(void *vpeddsactx, unsigned char *sigret,
     }
 #ifdef S390X_EC_ASM
     if (S390X_CAN_SIGN(ED25519)) {
-	    if (s390x_ed25519_digestsign(edkey, sigret, tbs, tbslen) == 0) {
-		    ERR_raise(ERR_LIB_PROV, PROV_R_FAILED_TO_SIGN);
-		    return 0;
-	    }
-	    *siglen = ED25519_SIGSIZE;
-	    return 1;
+        if (peddsactx->dom2_flag
+                || peddsactx->context_string_flag
+                || peddsactx->context_string_len > 0) {
+            /* s390x_ed25519_digestsign() does not yet support dom2 or context-strings */
```

Comment:
I considered that, but I didn't want users to puzzle over a mysterious slow down in EdDSA performance on s390x (e.g. "why is `Ed25519ph` slower than `Ed25519` on s390x?").  Having the call fail provides more immediate feedback to the user (and possibly provides motivation for someone to get this working on s390x).

I don't have a strong opinion here.  Maybe it is fine to do as you suggest.

I guess I can add a new comment in the code that devs can point to later on when someone notices the performance drop.

Reply:
I've just pushed a new revision with this change.

Reply:
I don't think the proposed behaviour works to give 'immediate feedback'. If someone ships an application which uses these algorithms, it will suddenly break when porting to s390x. If we support an algorithm on one platform we should be supporting it on all platforms, regardless of the performance we can offer.